feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -365,3 +388,169 @@ fun CaptureButton(
         })
     }
 }
+
+enum class ToggleState {
+    Left,
+    Right
+}
+
+@Composable
+fun ToggleButton(
+    leftIcon: Painter,
+    rightIcon: Painter,
+    modifier: Modifier = Modifier.width(64.dp).height(32.dp),
+    initialState: ToggleState = ToggleState.Left,
+    onToggleStateChanged: (newState: ToggleState) -> Unit = {},
+    isDisabled: Boolean = false,
+    iconPadding: Dp = 8.dp
+) {
+    val backgroundColor = MaterialTheme.colorScheme.surface
+    val disableColor = MaterialTheme.colorScheme.onSurface
+    val iconSelectionColor = MaterialTheme.colorScheme.onPrimary
+    val iconUnSelectionColor = MaterialTheme.colorScheme.primary
+    val circleSelectionColor = MaterialTheme.colorScheme.primary
+    val circleColor = if (isDisabled) disableColor.copy(alpha = 0.12f) else circleSelectionColor
+    val density = LocalDensity.current
+    var width by remember { mutableStateOf(0.dp) }
+    var height by remember { mutableStateOf(0.dp) }
+    var toggleState by remember { mutableStateOf(initialState) }
+    val animatedTogglePosition by animateFloatAsState(
+        when (toggleState) {
+            ToggleState.Left -> 0f
+            ToggleState.Right -> 1f
+        },
+        label = "anchor"
+    )
+    val scope = rememberCoroutineScope()
+
+    Surface(
+        modifier = modifier
+            .onSizeChanged {
+                width = with(density) { it.width.toDp() }
+                height = with(density) { it.height.toDp() }
+            }
+            .clip(shape = RoundedCornerShape(50))
+            .background(backgroundColor)
+            .then(
+                if (isDisabled) {
+                    Modifier
+                } else {
+                    Modifier.clickable {
+                        scope.launch {
+                            toggleState = when (toggleState) {
+                                ToggleState.Left -> ToggleState.Right
+                                ToggleState.Right -> ToggleState.Left
+                            }
+                            onToggleStateChanged(toggleState)
+                        }
+                    }
+                }
+            )
+    ) {
+        Box {
+            Row(
+                modifier = Modifier.matchParentSize(),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Box(
+                    Modifier
+                        .offset {
+                            IntOffset(
+                                x = with(density) {
+                                    animatedTogglePosition * (width - height).toPx()
+                                }.roundToInt(),
+                                y = 0
+                            )
+                        }
+                        .size(height)
```

Comment:
This can be replaced with the following code to eliminate the need for width/height variables:
```
                        .layout { measurable, constraints ->
                            val placeable = measurable.measure(constraints)
                            layout(placeable.width, placeable.height) {
                                val xPos = animatedTogglePosition *
                                        (constraints.maxWidth - placeable.width)
                                placeable.placeRelative(xPos.toInt(), 0)
                            }
                        }
                        .fillMaxHeight()
                        .aspectRatio(1f)
```

Reply:
Thank you, updated! It looks more straight forward now.